### ISSUE TITLE
Need a more elegant way to downscale an universe in VM/baremetal deployment

### ISSUE THREAD

**Issue by ricemaster** [yugabyte/yugabyte-db#5608]
Hi guys,

I'm looking for an elegent way to downscale an universe, but I failed.

From the docs on yb website, I learnt the best and safest way to downscale is to use blacklist. By executing `yb-admin -master_addresses [...] change_blacklist ADD <tserver-A>`, there would be no data on tserver-A any more after a while.

Now it acts like the universe has been downscaled, but it looks like not. Because this tserver-A would still be shown in the tserver list when executing `yb-admin -master_addresses [...] list_all_tablet_servers`.

So is there any other commands to do downscaling, like to transfer data to other nodes and remove from the universe for good?

Thanks.

**Comment by ddorian** [comment]
Hi @ricemaster 

Can you follow all the steps in https://github.com/yugabyte/yugabyte-db/pull/4790 ?

That is a subset of the more complex cluster config available in https://docs.yugabyte.com/latest/manage/change-cluster-config/

**Comment by rkarthik007** [comment]
cc @ameyb @stevebang - another one to document.

**Comment by ricemaster** [comment]
Hi @ddorian 

Thanks for reply.
I've read the issue and docs you referred. Here's what I got.

First of all, I have a universe with 7 tserver nodes. This is the result when running `list_all_tablet_servers`.
<img width="1322" alt="截屏2020-09-09 09 52 56" src="https://user-images.githubusercontent.com/10449002/92545451-be43b500-f282-11ea-8aa6-6bd5249b7855.png">

Then, I want to remove the node `10.211.55.105` from this universe. So I use `change_blacklist ADD 10.211.55.105:9100`.
<img width="1323" alt="截屏2020-09-09 09 53 08" src="https://user-images.githubusercontent.com/10449002/92545824-943ec280-f283-11ea-8a26-e3fb6f15a581.png">

And I also checked the completion by `get_load_move_completion`.
<img width="1204" alt="截屏2020-09-09 09 53 36" src="https://user-images.githubusercontent.com/10449002/92545996-08796600-f284-11ea-9c32-3cf2518d3778.png">

At last, all data has been transferred, and this `10.211.55.105` is `DEAD`.
<img width="1301" alt="截屏2020-09-09 09 53 57" src="https://user-images.githubusercontent.com/10449002/92546152-7e7dcd00-f284-11ea-8e58-fc0d16001773.png">

So now what I really want to know is, how can it be NOT shown when executing `list_all_tablet_servers` any more?

Because from the business or higher perspective, this node is no longer the part of this universe.

**Comment by bmatican** [comment]
@ricemaster Good point. We've previously added a `--exclude_dead` flag to `yb-admin list_all_tablet_servers`, see this commit: 7959fde66103ab72d74b8e74ee072b0e4f45447e

We also have this issue left open for OSS contributors, to also not display them in the UI after some configurable amount of time: #2627

**Comment by ricemaster** [comment]
@bmatican `-exclude_dead` is better. 

But if my understanding is current, `DEAD` is quite a complicated status in yb. It means the heartbeat is stopped, means this tserver node cann't send/receive messages to/from the master nodes. 

So it hard to tell that a `DEAD` node is temporarily dead or permenitly. 
if this node is dead because of some network or hardware issues which means it need to be repared and back to work, then it should be shown in the list, and marked `DEAD` as a alert. 
But if this node is ready to be removed from this universe, then it may be better not to show it in the list.

In other word, it would be useful to have some `DEAD` nodes in the list to warn DBA there's an error node, meanwhile not list the other `DEAD` nodes which are removed from the cluster maybe months ago.
